packages/devtools_app/lib/src/logging/flutter/logging_screen.dart

```diff
@@ -34,16 +36,22 @@ class LoggingScreenBody extends StatefulWidget {
 }
 
 class _LoggingScreenState extends State<LoggingScreenBody> {
+  LoggingController controller;
+
   @override
-  void initState() {
-    super.initState();
-    // Enable structured errors by default as soon as the user opens the
-    // logging page.
-    serviceManager.serviceExtensionManager.setServiceExtensionState(
-      structuredErrors.extension,
-      true,
-      true,
-    );
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    controller?.onLogsUpdated?.unregister(this);
```

Comment:
@kenzieschmoll With a controller from the inherited widget, this is how we want to manage subscriptions.